Eval.hs

```diff
@@ -444,6 +452,22 @@ isIdPair :: Val -> Bool
 isIdPair VIdPair{} = True
 isIdPair _         = False
 
+idComp :: Val -> Val -> Val -> Val -> Val -> Val -> Val
+idComp a u v w pId qId = case (pId, qId) of
+  (VIdPair p ps, VIdPair q qs)          ->
+    let i:j:_ = freshs [a,u,v,w,pId,qId] in
+    let
+      rs = unionSystem
+        (unionSystem
+          (border (q @@ (Atom i :/\: Atom j)) ps)
+          (border (p @@ (Atom i :\/: NegAtom j)) qs))
+        (mkSystem [((i ~> 0), p @@ NegAtom j), ((i ~> 1), q @@ Atom j)])
+    in
+    VIdPair (VPLam i (comp j a v rs)) (joinSystem (border ps qs))
+  (VIdPair p ps, qId) | eps `member` ps -> qId
+  (pId, VIdPair q qs) | eps `member` qs -> pId
```

Comment:
I'm still a bit unsure about these two reduction rules. It seems dangerous to add them, but I don't have any concrete evidence why. 

@simhu : Maybe you have more thoughts about this?